showkase-processor/src/main/java/com/airbnb/android/showkase/processor/ShowkaseProcessor.kt

```diff
@@ -50,17 +52,31 @@ class ShowkaseProcessor @JvmOverloads constructor(
     private val logger = ShowkaseExceptionLogger()
     private val showkaseValidator = ShowkaseValidator()
 
-    override fun getSupportedAnnotationTypes(): MutableSet<String> = mutableSetOf(
-        ShowkaseComposable::class.java.name,
-        PREVIEW_CLASS_NAME,
-        ShowkaseColor::class.java.name,
-        ShowkaseTypography::class.java.name,
-        ShowkaseRoot::class.java.name,
-        ShowkaseScreenshot::class.java.name,
-    )
-
-    override fun getSupportedOptions(): MutableSet<String> {
-        return mutableSetOf("skipPrivatePreviews")
+    override fun getSupportedAnnotationTypes(): MutableSet<String> {
+        val supportedAnnotationTypes = mutableSetOf(
+            ShowkaseComposable::class.java.name,
+            PREVIEW_CLASS_NAME,
+            ShowkaseColor::class.java.name,
+            ShowkaseTypography::class.java.name,
+            ShowkaseRoot::class.java.name,
+            ShowkaseScreenshot::class.java.name,
+        )
+        supportedAnnotationTypes.addAll(supportedAnnotationTypes)
+        supportedAnnotationTypes.addAll(supportedCustomAnnotationTypes())
+        return supportedAnnotationTypes
+    }
+    override fun getSupportedOptions() = mutableSetOf("skipPrivatePreviews", "multiPreviewTypes")
+
+    // Getting the custom annotations that are supported as an compiler argument.
+    // It is expected to get the compiler argument as follows:
+    // arg("multiPreviewTypes", "com.airbnb.android.submodule.showkasesample.FontPreview")
+    private fun supportedCustomAnnotationTypes(): MutableSet<String> {
+        val set = mutableSetOf<String>()
+        environment
+            .options["multiPreviewTypes"]
```

Comment:
Should we add any kind of validation here?

Reply:
Yeah I agree. I removed this piece since it works with the codegen strategy :)